src/controller/osw-controller.ts

```diff
@@ -61,38 +93,56 @@ class GtfsOSWController implements IController {
+            const folderPath = storageService.getFolderPath(oswdto.tdei_org_id,uid);
+            const uploadedFile = request.file;
+            const uploadPath = path.join(folderPath,uploadedFile!.originalname)
+            const remoteUrl = await storageService.uploadFile(uploadPath,'application/zip',Readable.from(uploadedFile!.buffer))
+            // Upload the meta file  
+            const metaFilePath = path.join(folderPath,'meta.json');
+            const metaUrl = await storageService.uploadFile(metaFilePath,'text/json',oswdto.getStream());
+            // Insert into database
+            const osw = OswVersions.from(meta);
+            osw.tdei_record_id = uid;
+            osw.file_upload_path = remoteUrl;
+            osw.uploaded_by = userId;
+            const returnInfo = await oswService.createOsw(osw);
+
+            // Publish to the topic
+            this.eventBusService.publishUpload(oswdto,uid,remoteUrl,userId,metaUrl);
+            // Also send the information to the queue
+            console.log('Responding to request');
+            return response.status(200).send(uid);
 
-            return validate(osw).then(async errors => {
-                // errors is an array of validation errors
-                if (errors.length > 0) {
-                    console.error('osw metadata information failed validation. errors: ', errors);
-                    const message = errors.map((error: ValidationError) => Object.values(<any>error.constraints)).join(', ');
-                    response.status(400).send('Input validation failed with below reasons : \n' + message)
-                    next(new HttpException(400, 'Input validation failed with below reasons : \n' + message));
-                } else {
-                    return await oswService.createOsw(osw)
-                        .then(newOsw => {
-                            return Promise.resolve(response.status(200).send(newOsw));
-                        })
-                        .catch((error: any) => {
-                            if (error instanceof DuplicateException) {
-                                response.status(error.status).send(error.message)
-                                next(new HttpException(error.status, error.message));
-                            }
-                            else {
-                                response.status(500).send('Error saving the osw version')
-                                next(new HttpException(500, 'Error saving the osw version'));
-                            }
-                        });
-                }
-            });
         } catch (error) {
-            console.error('Error saving the osw version', error);
-            response.status(500).send('Error saving the osw version')
-            next(new HttpException(500, "Error saving the osw version"));
+            console.error('Error saving the osw file', error);
+            response.status(500).send('Error saving the osw file');
```

Comment:
Please check the code from other repo. It should respond with `next` if the `error` is of type `HTTPException`

Reply:
fixed